src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -1123,11 +1133,27 @@ private int GetNumSamples(string path)
 
             trainSaver.save(_session, _checkpointPath);
             UpdateTransferLearningModelOnDisk(_classCount);
+            TryCleanupTemporaryWorkspace();
+        }
+
+        private void TryCleanupTemporaryWorkspace()
+        {
+            if (_cleanupWorkspace && Directory.Exists(_options.WorkspacePath))
```

Comment:
Assuming you hold the file pointers open, an elegant method is using the `FileOptions.DeleteOnClose` when you open the temporary files. 

This then causes the OS (or filesystem when on macOS/Linux) to cleanup the files when they are closed, or if your program crashes or is killed by the user.

https://docs.microsoft.com/en-us/dotnet/api/system.io.fileoptions?view=netcore-3.0